mlflow/types/llm.py

```diff
@@ -305,6 +305,12 @@ def __post_init__(self):
                 f"Expected `usage` to be of type TokenUsageStats or dict, got {type(self.usage)}"
             )
 
+    def __setattr__(self, name, value):
```

Comment:
Could a property work here as well? 

```python
    id: str
    model: str
    choices: List[ChatChoice]
    usage: TokenUsageStats
    _object: str = field(init=False, repr=False, default="chat.completion")
    created: int = field(default_factory=lambda: int(time.time()))

    @property
    def object(self):
        return self._object
```

The exception would change to a more generic `AttributeError` with a "can't set attribute" message, but it might help to avoid hijacking the attribute setter dunder method. 

WDYT?

Reply:
I think that suggestion makes sense, if we want to throw a more explicit error i think it can be possible by modifying the setter:

```python
@object.setter
def object(self, value):
  if value != "chat.completion":
    raise Exception("`object` must be 'chat.completion' and cannot be overridden")
  self._object = value
```

Reply:
both suggestions make sense to me.

but actually using property was the first approach I tried and it somehow didn't work hence I went with setattr. The reason it didn't work was because I tried to modify the @object.setter together as what @daniellok-db suggested. the setter would just not work.

interestingly the change suggested by @BenWilson2 above (only defining it as property without touching other part) worked. have updated the PR. thank you both!

python is hard T_T

Reply:
Hmm, what are the cases where it doesn't work? I tested briefly with adding the setter above and i got the following:

```python
# when trying to set object in `init`
>>> a = ChatResponse(id="hello", model="test", choices=[], usage=TokenUsageStats(1,1,1), object="123")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'object'

# when setting object after init
>>> a = ChatResponse(id="hello", model="test", choices=[], usage=TokenUsageStats(1,1,1))
>>> a.object = "123"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/daniel.lok/mlflow/mlflow/types/llm.py", line 302, in object
    raise AttributeError("`object` must be chat.completion")
AttributeError: `object` must be chat.completion
```

Reply:
do you have both _object and object defined? for my case it's the default value couldn't be passed in during init

another weird thing is that touching setter only disable response.object but in many unit tests I see people access it by response["object"] in a dict manner. that would also cause problems

Reply:
Also let's double check `to_dict()` / [asdict()](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) works as expected when we add the private `_object` field. Only the public `object` field should be included in the serialized format.

Reply:
Oh good point, we might have to edit the `to_dict()` method with a special case...

@xq-yin i think it should only be possible to do `response["object"]` after `.to_dict()` has been called, otherwise there should be an error like `TypeError: 'ChatResponse' object is not subscriptable`

Reply:
Reverting back to original __setattr__ solution and requesting review. Rationale:

solution 1:
`_object: str = field(init=False, repr=False, default="chat.completion") `
and 
```
@property 
def object(self): 
return self._object
```
this approach doesn't work when users try to set 'object' property when trying to initialize the ChatResponse instance
e.g.
```
mock_response["object"] = "chat.completion"
response = ChatResponse(**mock_response)
```
got error 
`__init__() got an unexpected keyword argument 'object'`

which requires that in addition to _object, object field also needs to be defined

solution 2:
when both object and _object are defined, we would need to touch the setter anyway
```
@object.setter
def object(self, value, key):
check and throw
```
plus I actually couldn't make this work with rounds of debugging

HENCE - going with modifying the __setattr__ given it's the easiest way to make the change and probably not that different to touching object. setter, apart from that the setter is a bit more generic

let me know what everyone thinks